src/modules/map/components/toolbox/TimeSlider.vue

```diff
@@ -49,12 +49,56 @@ const layersWithTimestamps = computed(() =>
 )
 const activeLayers = computed(() => store.state.layers.activeLayers)
 const previewYear = computed(() => store.state.layers.previewYear)
-const invalidYear = computed(
-    () =>
-        displayedYear.value != previewYear.value &&
-        displayedYear.value.toString().length == 4 &&
-        !allYears.value.includes(parseInt(displayedYear.value))
-)
+
+const isInputYearValid = ref(true)
+
+/**
+ * Used for the year in the input field Validate the input from the user. In case it's invalid, we
+ * don't propagate the value to the state, but instead save it to an intermediate state variable to
+ * be displayed along with an error message. This is important so that the cursor doesn't jump
+ * around to invalid years
+ */
+const inputYear = computed({
+    get() {
+        return falseYear.value || currentYear.value
+    },
+    set(value) {
+        if (value.length < 4) return value
```

Comment:
From our guideline we always use `if()` with brackets
```javascript
if (...) {
   ...
}
```

Reply:
Moreover this is an issue, if you enter a too small number, like 200, the validation is not done and you don't have a user feedback, even if you press enter.

Reply:
Damn. Good Catch.

Reply:
Althought we want to display the error only when the user finished entering... otherwise, say you start entering 184 you already get an error, but in case you wanted to type 1845, which would be OK, then the error is too early 🤔

Reply:
Do the validation on out of focus or when the user press enter 😉

Reply:
See TextInput component on how validation is done, maybe we could reuse this component.